src/components/TodoFooter.tsx

```diff
@@ -0,0 +1,99 @@
+import cn from 'classnames';
+import { useContext } from 'react';
+import { Todo } from '../types/Todo';
+import { deleteTodo } from '../api/todos';
+import { SetErrorContext } from '../utils/setErrorContext';
+import { ErrorMessage } from '../utils/ErrorMessage';
+
+interface Props {
+  setFilteringMode: (arg0: string) => void,
+  filteringMode: string;
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setTodosToBeEdited: React.Dispatch<React.SetStateAction<number[] | null>>,
+}
+
+export const TodoFooter: React.FC<Props>
+  = ({
+    setFilteringMode, filteringMode, todos, setTodosToBeEdited, setTodos,
+  }) => {
+    const setError = useContext(SetErrorContext);
+
+    const handleMassDeletion = () => {
+      const completedTodos = todos.filter(todo => todo.completed);
+
+      setTodosToBeEdited(completedTodos.map(todo => todo.id));
+
+      Promise.all(completedTodos.map((todo) => {
+        return new Promise((resolve) => deleteTodo(todo.id)
+          .then(resolve))
+          .catch(() => {
+            setTodosToBeEdited([]);
+            setError?.(ErrorMessage.CantDelete);
+          });
+      }))
+        .then(() => {
+          setTodosToBeEdited([]);
+          setTodos(todos.filter(todo => !todo.completed));
+        });
```

Comment:
you can put this inside `finally` instead of another then, also set error inside finally block

Reply:
not sure how exactly am I supposed to write that.

move `setTodosToBeEdited([])` to a `finally` block? that makes sense, but what about the error then? doesn't it need to be inside of `catch` to, well, trigger on an error instead of always?

what I'm thinking is
```tsx
      Promise.all(completedTodos.map((todo) => {
        return new Promise((resolve) => deleteTodo(todo.id)
          .then(resolve))
          .catch(() => {
            setError?.(ErrorMessage.CantDelete);
          });
      }))
        .then(() => {
          setTodos(todos.filter(todo => !todo.completed));
        })
      .finally(setTodosToBeEdited([]));
```